e2e/utils.go

```diff
@@ -1378,6 +1383,149 @@ func waitForJobCompletion(c kubernetes.Interface, ns, job string, timeout int) e
 	})
 }
 
+func waitForDeploymentUpdateScale(
+	c kubernetes.Interface,
+	ns,
+	deploymentName string,
+	scale *autoscalingv1.Scale,
+	timeout int) error {
+	t := time.Duration(timeout) * time.Minute
+	start := time.Now()
+	err := wait.PollImmediate(poll, t, func() (bool, error) {
+		scaleResult, upsErr := c.AppsV1().Deployments(ns).UpdateScale(context.TODO(),
+			deploymentName, scale, metav1.UpdateOptions{})
+		if upsErr != nil {
+			if isRetryableAPIError(upsErr) {
+				return false, nil
+			}
+			e2elog.Logf(
+				"Deployment UpdateScale %s/%s has not completed yet (%d seconds elapsed)",
+				ns, deploymentName, int(time.Since(start).Seconds()))
+
+			return false, fmt.Errorf("error update scale deployment %s: %w", deploymentName, upsErr)
+		}
+		if scaleResult.Spec.Replicas != scale.Spec.Replicas {
+			e2elog.Logf("scale result not matching for deployment %s, desired scale %d, got %d",
+				deploymentName, scale.Spec.Replicas, scaleResult.Spec.Replicas)
+
+			return false, fmt.Errorf("error scale not matching in deployment %s: %w", deploymentName, upsErr)
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed update scale deployment %s: %w", deploymentName, err)
+	}
+
+	return nil
+}
+
+func waitForDeploymentUpdate(
+	c kubernetes.Interface,
+	ns,
+	deploymentName string,
+	deployment *appsv1.Deployment,
+	timeout int) error {
+	t := time.Duration(timeout) * time.Minute
+	start := time.Now()
+	err := wait.PollImmediate(poll, t, func() (bool, error) {
+		_, upErr := c.AppsV1().Deployments(ns).Update(context.TODO(), deployment, metav1.UpdateOptions{})
+		if upErr != nil {
+			if isRetryableAPIError(upErr) {
+				return false, nil
+			}
+			e2elog.Logf(
+				"Deployment Update %s/%s has not completed yet (%d seconds elapsed)",
+				ns, deploymentName, int(time.Since(start).Seconds()))
+
+			return false, fmt.Errorf("error updating deployment %s: %w", deploymentName, upErr)
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed update deployment %s: %w", deploymentName, err)
+	}
+
+	return nil
+}
+
+func waitForClusterNameUpdate(c kubernetes.Interface, ns, deploymentName string, timeout int) error {
+	e2elog.Logf("waiting for deployment updates %s/%s", ns, deploymentName)
+
+	// Scale down to 0.
+	scale, err := c.AppsV1().Deployments(ns).GetScale(context.TODO(), deploymentName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error get scale deployment %s: %w", deploymentName, err)
+	}
+	count := scale.Spec.Replicas
+	scale.ResourceVersion = "" // indicate the scale update should be unconditional
+	scale.Spec.Replicas = 0
+	err = waitForDeploymentUpdateScale(c, ns, deploymentName, scale, timeout)
+	if err != nil {
+		return err
+	}
+
+	// Update deployment.
+	deployment, err := c.AppsV1().Deployments(ns).Get(context.TODO(), deploymentName, metav1.GetOptions{})
```

Comment:
remove the creationTimestamp,generation,resourceVersion, and uid in the metadata and try update?

Reply:
Done now, lets wait for CI